substrate/client/db/src/lib.rs

```diff
@@ -1707,7 +1708,8 @@ impl<Block: BlockT> Backend<Block> {
 							&(start, end).encode(),
 						);
 					}
-				} else if number > best_num + One::one() &&
+				} else if operation.create_gap &&
+					number > best_num + One::one() &&
```

Comment:
`number > best_num + One::one()` imply `number > One::one()`. We can eliminate the latter

Reply:
I eliminated the former instead to make it equivalent to the old code. Unless `number > One::one()` was a mistake in the first place.

Reply:
UPD: I see, agree, updated accordingly